File: java/pe/src/main/java/ktn/pe/Pe2.java
package ktn.pe;

import ktn.pe.util.PeUtils;

import java.util.logging.Logger;

public class Pe2 implements Pe {
    private static final Logger logger = Logger.getLogger(Pe2.class.getName());
    
    private long limit = 4000000;

    public long pe2(long limit) {
        long sum = 2;
        long a = 1;
        long b = 2;
        long t;

        while (b <= limit) {
            t = b;
            b += a;
            logger.fine(String.valueOf(b));
            a = t;
            if ((b & 1) == 0) {
                sum += b;
            }
        }

        return sum;
    }

    @Override
    public void setArgs(String[] args) {
        if (args.length > 0) {
            limit = Long.parseLong(args[0]);
        }
    }

    @Override
    public void solve() {
        System.out.println(PeUtils.format(this, pe2(limit)));
    }

    @Override
    public void run() {
        solve();
    }

    @Override
    public int getProblemNumber() {
        return 2;
    }
}

File: kt/pe/src/main/kotlin/ktn/pe/Pe2.kt
package ktn.pe

import java.util.logging.Logger

class Pe2 : Pe {

    private var limit: Long = 4000000

    override val problemNumber: Int
        get() = 2

    fun pe2(limit: Long): Long {
        var sum: Long = 2
        var a: Long = 1
        var b: Long = 2
        var t: Long

        while (b <= limit) {
            t = b
            b += a
            logger.fine(b.toString())
            a = t
            if (b and 1 == 0L) {
                sum += b
            }
        }

        return sum
    }

    override fun setArgs(args: Array<String>) {
        if (args.isNotEmpty()) {
            limit = args[0].toLong()
        }
    }

    override fun solve() {
        System.out.println(PeUtils.format(this, pe2(limit)))
    }

    override fun run() {
        solve()
    }

    companion object {
        private val logger = Logger.getLogger(Pe2::class.java.name)
    }
}
